src/validation.cpp

```diff
@@ -5009,26 +5129,54 @@ void ChainstateManager::Unload()
 
 void ChainstateManager::Reset()
 {
-    LOCK(::cs_main);
+    auto get_storage_path = [](auto& chainstate) EXCLUSIVE_LOCKS_REQUIRED(::cs_main) -> std::optional<fs::path> {
```

Comment:
This annotation in the definition is redundant, as it is already in the declaration.